src/NHibernate/Linq/GroupJoin/GroupJoinAggregateDetectionVisitor.cs

```diff
@@ -63,10 +63,9 @@ protected override Expression VisitMember(MemberExpression expression)
 
 		protected override Expression VisitQuerySourceReference(QuerySourceReferenceExpression expression)
 		{
-			var fromClause = (FromClauseBase) expression.ReferencedQuerySource;
-
-			var querySourceReference = fromClause.FromExpression as QuerySourceReferenceExpression;
-			if (querySourceReference != null)
+			if (!(expression.ReferencedQuerySource is FromClauseBase fromClause))
+			{ }
```

Comment:
That is strange formatting :-/

Reply:
I still don't get it why... Why not simply invert the if statement?

Reply:
To minimize change noize... Otherwise additional if statement block is required..
```C#
if ((expression.ReferencedQuerySource is FromClauseBase fromClause))
{
   if (fromClause.FromExpression is QuerySourceReferenceExpression querySourceReference)
   ...
   else if (fromClause.FromExpression is SubQueryExpression subQuery)
   ...
}
```

Reply:
I would prefer clearer code over minimizing the changes.